tests/rbd_mirror/test_rbd_mirror_shrink_image_primary.py

```diff
@@ -0,0 +1,131 @@
+        else:
+            mirror2.wait_for_status(imagespec=imagespec, state_pattern="up+replaying")
+
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+        out1 = rbd1.image_info(pool, image)
+        out2 = rbd2.image_info(pool, image)
+        if out1["size"] == out2["size"]:
+            log.info(f"New image size {resize} is updated in secondary")
+            return 0
+        log.error(f"New image size {resize} is not updated in secondary")
+        return 1
+
+    except Exception as e:
+        log.exception(e)
+        return 1
+
+    finally:
+        mirror1.clean_up(peercluster=mirror2, pools=[pool])
+
+
+def run(**kw):
+    log.info("Starting RBD mirroring test case - CEPH-9499")
+    """Shrink images on local/primary cluster while mirroring is happening.
+
+    Args:
+        **kw: test data
+        Example::
+            ec_pool_config:
+                mode: image
+                mirrormode: snapshot
+                size: 2G
+                io_total: 10
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+
+    Test case covered -
+    CEPH-11416 - Pool Mirror - Try marking the secondary on the relationship for deletion, and observe the behaviour
+
+    Pre-requisites :
+    1. At least two clusters must be up and running with enough number of OSDs to create pools
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Bring up 2 cluster and configure the mirroring.
+    2. run the IO on images.
+    3. Shrink the image on primary cluster by using --allow-shrink flag.
+    4. verify for the image status in secondary and verify for the size is updated.
+    """
+    try:
+        config = kw["config"]["journal"]
+        kw["config"].update(config)
+        mirror_obj = rbd_mirror_config(**kw)
+        if mirror_obj:
+            log.info("Executing test on replicated pool")
+            if test_image_shrink_primary(
+                mirror_obj.get("rep_rbdmirror"), "rep_pool_config", **kw
+            ):
+                return 1
+
+            log.info("Executing test on ec pool")
+            if test_image_shrink_primary(
+                mirror_obj.get("ec_rbdmirror"), "ec_pool_config", **kw
+            ):
+                return 1
+
+        config = kw["config"]["snapshot"]
+        kw["config"].update(config)
+        mirror_obj = rbd_mirror_config(**kw)
+        if mirror_obj:
+            log.info("Executing test on replicated pool")
+            if test_image_shrink_primary(
+                mirror_obj.get("rep_rbdmirror"), "rep_pool_config", **kw
+            ):
+                return 1
+
+            log.info("Executing test on ec pool")
+            if test_image_shrink_primary(
+                mirror_obj.get("ec_rbdmirror"), "ec_pool_config", **kw
+            ):
+                return 1
```

Comment:
It can be observed that there are similar lines of code
Can we make use of iterator ?

Reply:
LGTM